lib/collectors/global_prometheus_collector.rb

```diff
@@ -0,0 +1,40 @@
+require "prometheus_exporter"
+require "prometheus_exporter/server"
+
+# Load Rails models as not automatically included for Prometheus Exporter
+require File.expand_path("../../config/environment", __dir__) unless defined? Rails
+
+module Collectors
+  class GlobalPrometheusCollector < PrometheusExporter::Server::TypeCollector
+    def type
+      "app_global"
+    end
+
+    def metrics
+      token_expiry_timestamp = PrometheusExporter::Metric::Gauge.new("signon_api_user_token_expiry_timestamp_seconds", "Timestamp when API User token expires")
+
+      # Cache metric to prevent needless expensive calls to the database
+      token_expiry_info = Rails.cache.fetch("token_expiry_info", expires_in: 1.hour) do
+        ApiUser.all.flat_map do |user|
```

Comment:
Would it be possible to add some tests for this class? In particular, I'm worried that if we make changes to ApiUser/User that this would break without us realising.

Reply:
Added a simple test case to check the metrics report the correct tokens.